x-pack/test/fleet_api_integration/config.ts

```diff
@@ -66,6 +66,7 @@ export default async function ({ readConfigFile }: FtrConfigProviderContext) {
         `--xpack.fleet.packages.0.version=latest`,
         ...(registryPort ? [`--xpack.fleet.registryUrl=http://localhost:${registryPort}`] : []),
         `--xpack.fleet.developer.bundledPackageLocation=${BUNDLED_PACKAGE_DIR}`,
+        '--xpack.cloudSecurityPosture.enabled=true',
```

Comment:
@juliaElastic would love to get feedback if that is the way to go with it

Reply:
I'm not really familiar with `cloudSecurityPosture` config, are you able to test that it works?

Reply:
So the problem is that the `cloudSecurityPosture` plugin contains the SO type registration for `csp-rule-template` type and I'm guessing the package install fails when this SO type isn't known to Kibana because this plugin is currently disabled by default (still in development)

@kfirpeled is there an intention to release any packages that contain `csp-rule-template` objects before this plugin is enabled by default? If so, I think we need to move the SO type registration to a separate (probably temporary) plugin that is always enabled. I don't want there to be a situation where a user can install a package in our UI but it fails with some obscure error message.

Reply:
Your guess is correct @joshdover .
We plan to release the package only after we will go publicly.

What happened here, is that we introduced a new asset type in the package-spec (`csp-rule-template` SO). So we thought it would fit to add tests to the current `fleet_api install uninstall` test. 

@juliaElastic this config is what enables our plugin. We use this as a feature flag for our cloud security posture use case. The planned release is **8.3**. Which means this would be a dead code and we don't plan it would be available to use till then. 

So my followup question from this, if we took the wrong turn, adjusting the current tests in the fleet api. And if in our current situation, it is more suitable to add the tests someplace else (in our plugin alone). 
Would love to hear your thoughts and if you have better ideas for us to add such tests when introducing new asset type in the spec.

Reply:
> We plan to release the package only after we will go publicly.

Are we expecting users to be able to test this starting in 8.2? If so, how will users install the package if we don't release the package (as experimental)? Today, the UI shows all experimental packages to all users, so even users without the csp plugin enabled will see this. I think we need to handle this more gracefully, like with moving the SO registration to another plugin for now.

> So my followup question from this, if we took the wrong turn, adjusting the current tests in the fleet api. And if in our current situation, it is more suitable to add the tests someplace else (in our plugin alone).
Would love to hear your thoughts and if you have better ideas for us to add such tests when introducing new asset type in the spec.

I think if the Fleet code supports installing these asset types then the test should be in the Fleet tests, whether the code is dead or not. If we're not going to release even an experimental package in 8.2, then I'm fine with moving forward with the current change.

Reply:
1. We don't expect users to test this on 8.2.
Releasing the package would be the last part when rolling this out.
The prerequisites are: cloud-security-plugin in kibana, cloudbeat available for the agent. 
So if a user would try to enable it in 8.2 (in the cloud it is blocked since this configuration is not allowed to be used) - the effect would be registering two SO, api's being opened, that can't change anything since the package is not released, index pattern being created. 

2. So we can leave it as is. So does it makes sense enabling the plugin the way we did here for the tests? or is there a way achieve that in the scope the test preparation (beforeTest).

The configuration is the way for us to enable the plugin. without it, it is not loaded into kibana server.

Reply:
I think enabling the plugin like this in config looks good, I'm not sure if there is a way to only enable the plugin in test preparation.